vcd/resource_vcd_nsxt_security_group_test.go

```diff
@@ -0,0 +1,477 @@
+					resource.TestCheckResourceAttr("vcd_nsxt_security_group.group1", "name", "test-security-group"),
+					resource.TestCheckResourceAttr("vcd_nsxt_security_group.group1", "description", "test-security-group-description"),
+					resource.TestCheckTypeSetElemNestedAttrs("vcd_nsxt_security_group.group1", "member_vms.*", map[string]string{
+						"vm_name":   "vapp-vm",
+						"vapp_name": "web",
+					}),
+					resource.TestCheckTypeSetElemNestedAttrs("vcd_nsxt_security_group.group1", "member_vms.*", map[string]string{
+						"vm_name":   "standalone-VM",
+						"vapp_name": "",
+					}),
+				),
+			},
+			resource.TestStep{
+				Config: configText1,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestMatchResourceAttr("vcd_nsxt_security_group.group1", "id", regexp.MustCompile(`^urn:vcloud:firewallGroup:.*$`)),
+					resource.TestCheckResourceAttr("vcd_nsxt_security_group.group1", "name", "test-security-group"),
+					resource.TestCheckResourceAttr("vcd_nsxt_security_group.group1", "description", "test-security-group-description"),
+					resource.TestCheckTypeSetElemNestedAttrs("vcd_nsxt_security_group.group1", "member_vms.*", map[string]string{
+						"vm_name":   "vapp-vm",
+						"vapp_name": "web",
+					}),
+					resource.TestCheckTypeSetElemNestedAttrs("vcd_nsxt_security_group.group1", "member_vms.*", map[string]string{
+						"vm_name":   "standalone-VM",
+						"vapp_name": "",
+					}),
+					// Ensure datasource has all the fields
+					resourceFieldsEqual("vcd_nsxt_security_group.group1", "data.vcd_nsxt_security_group.group1", []string{}),
+				),
+			},
+			resource.TestStep{
+				Config: configText2,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestMatchResourceAttr("vcd_nsxt_security_group.group1", "id", regexp.MustCompile(`^urn:vcloud:firewallGroup:.*$`)),
+					resource.TestCheckResourceAttr("vcd_nsxt_security_group.group1", "name", "test-security-group-changed"),
+					resource.TestCheckResourceAttr("vcd_nsxt_security_group.group1", "description", ""),
+					resource.TestCheckNoResourceAttr("vcd_nsxt_security_group.group1", "member_org_network_ids"),
+					resource.TestCheckNoResourceAttr("vcd_nsxt_security_group.group1", "member_vm_ids"),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      "vcd_nsxt_security_group.group1",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: importStateIdNsxtEdgeGatewayObject(testConfig, testConfig.Nsxt.EdgeGateway, "test-security-group-changed"),
+			},
+		},
+	})
+	postTestChecks(t)
+}
+
+const testAccNsxtSecurityGroupPrereqs = testAccNsxtSecurityGroupPrereqsEmpty + `
+resource "vcd_network_routed_v2" "nsxt-backed" {
+	# This value could be larger to test more members, but left 2 for the sake of testing speed
+	count = 2
+
+	org  = "{{.Org}}"
+	vdc  = "{{.NsxtVdc}}"
+	name        = "nsxt-routed-${count.index}"
```

Comment:
This script is nor well formatted

Reply:
Reformatted all scripts with `terraform fmt` in this file.